### ISSUE TITLE
Using Toolchain in an IDE

### ISSUE THREAD

**Issue by sthomasbradley** [CE-Programming/toolchain#422]
You state "If you know how to configure the toolchain to work with another IDE, raise an issue or pull request on [GitHub](https://github.com/CE-Programming/toolchain)."

I know how to configure the toolchain to work with the Code::Blocks IDE.  It also can be set up to automatically load the finished program into CEmu and run it.  If you are interested email me and I'll send you the procedure.

**Comment by commandblockguy** [comment]
That would be a good thing to include in our docs, would you mind posting the procedure here?

**Comment by mateoconlechuga** [comment]
Yes, please add the documentation to this issue or send me an email. Thanks!

**Comment by mateoconlechuga** [comment]
I will close this issue in a week if no more information is provided.

**Comment by sthomasbradley** [comment]
[CodeBlocks.txt](https://github.com/CE-Programming/toolchain/files/10436648/CodeBlocks.txt)
[Visual Studio.txt](https://github.com/CE-Programming/toolchain/files/10436649/Visual.Studio.txt)
CodeBlocks.txt document fully integrates the Toolchain and CEmu into the CodeBlocks IDE.  The VisualStudio,txt is for a novice who wants to use an IDE,

**Comment by sthomasbradley** [comment]
In the CodeBlocks.txt instructions I posted yesterday I left out one important instruction.  In the makefile the NAME = must be the same as the CodeBlocks Title i.e. Title: DEMO, project name DEMO.cbp and NAME = DEMO.  If it isn't the same CEmu will throw the error "Transfer Error: see console for information".

**Comment by vrcdrafter** [comment]
My Make file only has this in it , 
What do I change ? 
#-----------------------------------------------
# This makefile builds all the example projects
#-----------------------------------------------

MAKEFILES = hello_world/makefile  $(wildcard */*/makefile) $(wildcard */*/*/makefile)

all clean gfx .PHONY: $(MAKEFILES)

$(MAKEFILES):
	$(MAKE) -C $(dir $@) $(or $(MAKECMDGOALS),$(findstring debug,$@))

.PHONY: all clean gfx

**Comment by adriweb** [comment]
You shouldn't have to change anything provided by the toolchain. 

You can/should however duplicate a folder of one example you choose as your starting point for a new project of yours, put it in another directory (for instance your desktop), and use "make" etc from there.

**Comment by vrcdrafter** [comment]
ah I got it , I chose the folder for the hello world example and I'm using that as my Codeblocks template . I'm all good now . Tool chain is fine , I just didnt know what I was doing .

**Comment by sthomasbradley** [comment]
[CodeBlocksBonus.txt](https://github.com/CE-Programming/toolchain/files/10700800/CodeBlocksBonus.txt)
Here is a "Bonus" note I put together for using the Toolchain and CEmu with the CodeBlocks IDE.  I hope someone may find it useful.

**Comment by vrcdrafter** [comment]
Im kind of a Dull pencil but is anybody else getting this issue on the compile ? Its like the compiler is trying to run the output file .8xp which I don't want at all . 
![image](https://user-images.githubusercontent.com/102326953/231475892-d8aaf722-3202-45f6-b39b-4205377effbf.png)

**Comment by adriweb** [comment]
Well yes, don't try to debug/run it (or, if you do, set that up correctly to make it use CEmu with the CLI option to transfer&launch the 8xp).

In addition, you might want to remove the zipping part (not sure why that is there, but why not) or make "zip" available in your path, if that is the problem (just a theory)

**Comment by sthomasbradley** [comment]
Both the zip and CEmu commands in the post-build section of the CodeBlocks project are command line instructions to your computer.  The parts that include $ are parts CodeBlock fills in before sending the command to your computer.  Both are meant to be written the way I wrote them or they may not work properly.
The "zipping part" is completely optional and can be removed. It would create a new zip either daily ($TODAY) or every time you compiled ($NOW).  That way you have a backup of all your files.  I occasionally make such a mess of my code that I've found it handy to be able to simply unzip the backup and start over again.  
As Adriweb said you must have a standard zip program installed on your computer and it must run from the command line using the command "zip"
The command to run/debug that I included are CEmu commands and are necessary for CEmu to understand what you want it to do.
